src/Model.php

```diff
@@ -1888,9 +1888,41 @@ public function action($mode, $args = [])
      */
     public function toQuery($mode, $args = []): Persistence\AbstractQuery
     {
-        $this->checkPersistence('action');
+        $this->checkPersistence('query');
 
-        return $this->persistence->action($this, $mode, $args);
+        // start backward compatibility -->
```

Comment:
new way?

Reply:
Up for discussion - if and when to deprecate